src/Application/Model/Ticket.php

```diff
@@ -851,6 +851,33 @@ public function resetSource(Comment $comment = null) {
 			
 			return true;
 		}
+
+		public function reset($handle_id = null) {
+			$query = Database::$Instance->query(
+				'SELECT * FROM ticket_reset(?)',
+				[$this['id'], ]
+			);
+
+			$tickets = $query->fetchAll();
+			foreach ($tickets as $ticket) {
+				$entry = [
+					'ticket_id' => $ticket['ticket_id'],
+					'from_state' => $ticket['from_state'],
+					'to_state' => $tickets['to_state'],
+					'handle_id' => ($handle_id === null) ? User::getCurrent()['id'] : $handle_id,
+					'event' => 'Action.reset'
+					];
+
+				if ($ticket['follow_meta_ticket']) {
+					$entry['comment'] = 'Following meta ticket.';
```

Comment:
Mh, not sure about the phrasing here. Maybe "Reset due to meta ticket reset" or something similar is better? @jjeising ?